src/main/java/engine/entities/Board.java

```diff
@@ -142,6 +142,7 @@ public void advanceCurrentPlayer() {
      */
     public void setPieceAtAbsCoords(int x, int y, Piece piece) {
         board[y][x] = piece;
+        piece.setCoords(x, y);
```

Comment:
Not sure about this. I think the action of moving a piece should be abstracted elsewhere. The action of moving a piece (removing it from its old position, replacing it at the new position, then updating its coordinates) is comprised of three disparate parts. Directly calling piece.setCoords() requires us to know about the implementation of piece, while keeping the code elsewhere allows for looser coupling